### ISSUE TITLE
mkdir() tries to alter the path when using different systems (Linux / Mac and windows) between remote and local servers

### ISSUE THREAD

**Issue by wangtao2012** [steelbrain/node-ssh#76]
For example if I use a windows machine to run `ssh.mkdir('/abc')`, it will try to create `c:\abc` since the path had been altered by `makedirp`.

**Comment by steelbrain** [comment]
Hi @wangtao2012 

Thanks for reminding me. I've added the `mkdir()` command options in README. If you use the `sftp` method and the server has sftp enabled, you shouldn't have any program changing the way they are represented.

**Comment by andyhu** [comment]
Hi, the problem is not related to parameter, I've tried even with `sftp` parameter it still changes the path.
In `mkdirp` module there's a line of code like this:
```js
    p = path.resolve(p);
    
    xfs.mkdir(p, mode, function (er) {
```
If I run this script from windows with a path name like '/var/www', it will be resolved to 'c:\var\www', this is what `path.resolve` actually do. It's hardcoded in the function, so there's no way to get around it. Can we ditch `mkdirp` and write it within `node-ssh`?

**Comment by wangtao2012** [comment]
Hi, the problem is not related to the parameter, I've tried even with sftp parameter it still changes the path.
In mkdirp module there's a line of code like this:

    p = path.resolve(p);
    
    xfs.mkdir(p, mode, function (er) {
If I run this script from windows with a path name like '/var/www', it will be resolved to 'c:\var\www', this is what path.resolve actually does. It's hardcoded in the function, so there's no way to get around it. Can we ditch mkdirp and write the logic within node-ssh?

**Comment by joneal** [comment]
I just encountered this same issue.  (Client: Windows 10, Server: AWS Linux VM)

**Comment by steelbrain** [comment]
Sorry it's taken so long, I'll try to get that PR merged and the fix released today

**Comment by steelbrain** [comment]
I've published `node-ssh` v4.2.1 with the fix for this
